Content.Client/UserInterface/Controls/ActionSlot.cs

```diff
@@ -340,7 +340,8 @@ public void CancelPress()
         /// </summary>
         public void Depress(bool depress)
         {
-            if (!CanUseAction) return;
+            // Allow people to toggle repeatable actions on cooldown.
+            if (!CanUseAction && (!Action?.Repeat ?? true)) return;
```

Comment:
Your above changes made it so the action stays selected even when on cooldown, regardless of if it's repeatable or not, which IMO is fine. I think you should do that here as well - let it be toggleable if it's a targetpoint or targetentity action. As it stands, this would make you unable to untoggle a non-repeatable target action if for some reason it went on cooldown.